code/controllers/subsystem/statpanel.dm

```diff
@@ -50,63 +50,114 @@ SUBSYSTEM_DEF(statpanels)
 	while(length(currentrun))
 		var/client/target = currentrun[length(currentrun)]
 		currentrun.len--
-		var/ping_str = url_encode("Ping: [round(target.lastping, 1)]ms (Average: [round(target.avgping, 1)]ms)")
-		var/other_str = url_encode(json_encode(target.mob.get_status_tab_items()))
-		target << output("[encoded_global_data];[ping_str];[other_str]", "statbrowser:update")
+		if(!target.statbrowser_ready)
+			target << browse(file('html/statbrowser.html'), "window=statbrowser")
+			continue
+		if(target.stat_tab == "Status")
+			var/ping_str = url_encode("Ping: [round(target.lastping, 1)]ms (Average: [round(target.avgping, 1)]ms)")
+			var/other_str = url_encode(json_encode(target.mob.get_status_tab_items()))
+			target << output("[encoded_global_data];[ping_str];[other_str]", "statbrowser:update")
 		if(!target.holder)
 			target << output("", "statbrowser:remove_admin_tabs")
 		else
-			var/turf/eye_turf = get_turf(target.eye)
-			var/coord_entry = url_encode(COORD(eye_turf))
-			target << output("[mc_data_encoded];[coord_entry];[url_encode(target.holder.href_token)]", "statbrowser:update_mc")
-			var/list/ahelp_tickets = GLOB.ahelp_tickets.stat_entry()
-			target << output("[url_encode(json_encode(ahelp_tickets))];", "statbrowser:update_tickets")
-			if(!length(GLOB.sdql2_queries))
-				target << output("", "statbrowser:remove_sqdl2")
-			else
-				var/list/sqdl2A = list()
-				sqdl2A[++sqdl2A.len] = list("", "Access Global SDQL2 List", REF(GLOB.sdql2_vv_statobj))
-				var/list/sqdl2B = list()
+			if(target.stat_tab == "MC" || !("MC" in target.panel_tabs))
+				var/turf/eye_turf = get_turf(target.eye)
+				var/coord_entry = url_encode(COORD(eye_turf))
+				target << output("[mc_data_encoded];[coord_entry];[url_encode(target.holder.href_token)]", "statbrowser:update_mc")
```

Comment:
You could always use some deferred execution here.

`mc_data_encoded` should not even be getting calculated if nobody has an mc tab active, but it should only get calculated once if multiple people have it active.